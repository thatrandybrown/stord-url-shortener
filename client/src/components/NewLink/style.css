.notice {
  margin-top: 21px;
  display: flex;
  flex-direction: column;
  padding: 7px;
  border: 2px rgb(150, 40, 150) solid;
  border-radius: 1em;
}
