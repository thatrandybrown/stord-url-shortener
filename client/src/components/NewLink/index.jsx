import "./style.css";

const copyLink = (event) => {
  const selection = window.getSelection();
  selection.removeAllRanges();

  const range = document.createRange();
  range.selectNodeContents(document.querySelector(".shortLink"));
  selection.addRange(range);

  document.execCommand("Copy");
  selection.removeAllRanges();
};

const NewLink = ({ data }) =>
  data && data.target && data.shortCode ? (
    <section className="notice">
      <p>
        <a href={data.target}>{data.target}</a> is now{" "}
        <a
          className="shortLink"
          href={`${window.location.href}${data.shortCode}`}
        >
          {window.location.href}
          {data.shortCode}
        </a>
      </p>
      <button onClick={copyLink}>Copy!</button>
    </section>
  ) : null;

export default NewLink;
