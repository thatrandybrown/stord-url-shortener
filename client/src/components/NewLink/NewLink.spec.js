import { fireEvent, render } from "@testing-library/react";
import NewLink from "./";

test("renders the Data Display", () => {
  const { container } = render(
    <NewLink data={{ target: "https://www.google.com", shortCode: "abcdef" }} />
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("renders nothing without data", () => {
  const { container } = render(<NewLink />);
  expect(container.firstChild).toMatchSnapshot();
});

test("links to the original link", () => {
  const { getByText } = render(
    <NewLink data={{ target: "https://www.google.com", shortCode: "abcdef" }} />
  );
  expect(getByText("https://www.google.com").href).toBe(
    "https://www.google.com/"
  );
});

test("links to the shortLink", () => {
  const { getByText } = render(
    <NewLink data={{ target: "https://www.google.com", shortCode: "abcdef" }} />
  );
  expect(getByText("http://localhost/abcdef").href).toBe(
    "http://localhost/abcdef"
  );
});

test("copies the link when the button is clicked", () => {
  let clipboardText;
  const execCommand = jest.fn((cmd) => {
    if (cmd === "Copy") clipboardText = window.getSelection().toString();
  });
  global.document.execCommand = execCommand;

  const testData = { target: "https://www.google.com", shortCode: "abcdef" };
  const { getByRole } = render(<NewLink data={testData} />);

  const copyButton = getByRole("button", { name: "Copy!" });
  fireEvent.click(copyButton);

  expect(execCommand).toHaveBeenCalled();
  expect(clipboardText).toEqual("http://localhost/abcdef");
});
