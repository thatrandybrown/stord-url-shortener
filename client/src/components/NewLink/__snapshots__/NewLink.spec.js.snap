// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nothing without data 1`] = `null`;

exports[`renders the Data Display 1`] = `
<section
  class="notice"
>
  <p>
    <a
      href="https://www.google.com"
    >
      https://www.google.com
    </a>
     is now
     
    <a
      class="shortLink"
      href="http://localhost/abcdef"
    >
      http://localhost/
      abcdef
    </a>
  </p>
  <button>
    Copy!
  </button>
</section>
`;
